### ISSUE TITLE
Create Articles on Algorithms

### ISSUE THREAD

**Issue by alayek** [FreeCodeCamp/wiki#1048]
[Note: Not to confuse with our algorithm challenges in JavaScript]

We need solid articles on following Algorithm-related topics:

- [ ] Sorting
     - [ ] In-place sorting
     - [ ] In-order sorting
     - [ ] Bubblesort
     - [ ] Selection Sort
     - [ ] Insertion Sort
     - [ ] Merge Sort
     - [ ] Heap Sort
     - [ ] Quicksort
- [ ] Divide and Conquer
- [ ] Greedy Algorithms
- [ ] Dynamic Programming
      - [ ] Edit Distance
      - [ ] Matrix Multiplication
 - [ ] String Matching
       - [ ] Naive Matching
       - [ ] Rabin Karp
       - [ ] KMP

It should be made clear to the reader in the article, that for all of these, libraries exist, and that in production, they should use that. However, this can be helpful in a technical interview.

The implementation section should have language based subsections.

**Comment by alayek** [comment]
@varunu28 do you think we should add Big-O notation under this epic?

**Comment by varunu28** [comment]
@alayek  I think we should remove the Big-O notation article from DS and put it under here. There  would be a topic in the starting where the Big-O notation would be explained and then for every algorithm article we would have a small topic where we would explain Big-O notation for that algorithm.

**Comment by varunu28** [comment]
@alayek In the above issue what algorithms do we need to mention under in-place sorting and in-order sorting as algorithms like heap sort and quick sort are considered as in-place algorithms?
Do we need to just explain what is meant by in-place and in-order sorting?

**Comment by varunu28** [comment]
@alayek I want to get started with the bubble sort. Please assign that to me.

**Comment by alayek** [comment]
@varunu28 yes, you need to clearly mention what is meant by those terms, and explain with examples, why the examples satisfy those terms.

Assigned.

**Comment by jainaman224** [comment]
Please assign me insertion sort.

**Comment by alayek** [comment]
ok. done @jainaman224

**Comment by jainaman224** [comment]
Can you please see #1056 for insertion sort.

**Comment by varunu28** [comment]
@alayek I am done with BubbleSort. I will start with selection sort. Please assign that to me

**Comment by atjonathan** [comment]
@varunu28, added you!

**Comment by angregor** [comment]
I can start working on MergeSort, please assign that to me. Also are we still in need of a page on Big-O notation? If so, I'll work on that as opposed to MergeSort.

**Comment by angregor** [comment]
@atjonathan I believe MergeSort has already been pushed so it can probably be removed unless the current article isn't sufficient. Also, I'm going to begin working on an article for Big-O notation so please assign that one to me.

**Comment by atjonathan** [comment]
@angregor, thanks!

**Comment by alayek** [comment]
@angregor thanks for pointing out. That article requires some updating and modification with more code examples in Python/Java/C++.

We will update the existing one.

**Comment by dakshshah96** [comment]
Kindly update the list with #1172 for Binary Search.

**Comment by alayek** [comment]
@dakshshah96 done :+1:

**Comment by ilogin** [comment]
Please assign Searching - Naive Search to me.

**Comment by anirudherabelly** [comment]
I will write on Greedy algorithms. If you wish to! @alayek
